src/main/java/graphql/schema/PropertyDataFetcher.java

```diff
@@ -175,13 +177,47 @@ private Object getPropertyViaGetterUsingPrefix(Object object, String prefix, Met
 
     @SuppressWarnings("SimplifiableIfStatement")
     private boolean isBooleanProperty(GraphQLOutputType outputType) {
-        if (outputType == GraphQLBoolean) return true;
+        if (outputType == GraphQLBoolean) {
+            return true;
+        }
         if (outputType instanceof GraphQLNonNull) {
             return ((GraphQLNonNull) outputType).getWrappedType() == GraphQLBoolean;
         }
         return false;
     }
 
+    private static final ConcurrentMap<String, Method> METHOD_CACHE = new ConcurrentHashMap<>();
+    private static final ConcurrentMap<String, Field> FIELD_CACHE = new ConcurrentHashMap<>();
+
+    /**
+     * PropertyDataFetcher caches the methods and fields that map from a class -> a property for runtime performance reasons.
+     *
+     * However during development you might be using an assistance tool like JRebel to allow you to tweak your code base and this
+     * caching may interfere with this.  So you can call this method to clear the cache.  A JRebel plugin could
+     * be developer to do just that.
+     */
+    @SuppressWarnings("unused")
+    public static void clearReflectionCache() {
+        METHOD_CACHE.clear();
+        FIELD_CACHE.clear();
+    }
+
+    private String mkKey(Class clazz, String propertyName) {
+        return clazz.getCanonicalName() + "__" + propertyName;
```

Comment:
Maybe base the cache key on something else than clazz.getCanonicalName() which seems to be slow due to being a native method?

https://bugs.openjdk.java.net/browse/JDK-5058117

consider if getName() is sufficient, since it seems to be cached.

Reply:
also, getCanonicalName seems to return null for anonymous inner classes:

```java
package foo;
public class Foo {

    interface Bar {
        void baz();
    }

    public static void main(String[] args) {
        System.out.println("class name:"+Foo.class.getName());

        Bar b = new Bar() {
            @Override
            public void baz() { }
        };

        System.out.println("anon inner class name: "+ b.getClass().getName());
        System.out.println("anon inner class canonicalName: "+b.getClass().getCanonicalName());
    }

}
```

Reply:
Why not use a SimpleImmutableEntry as a stdlib included Pair/Tuple implementation for the key?

Reply:
Thanks - will change it to Class.getName()